docs/DeveloperGuide.md

```diff
@@ -1721,10 +1722,52 @@ The team has put in a tremendous amount of effort to this project, with a single
   codebase. In the following section, the effort will be further elaborated.
   
-### App's functionality 
+### App's functionality & User Experience
 
-When we first start the project, we were quite surprised that the app must be optimized for CLI, which is not a
- common thing for most of the commercial apps nowadays
+When we first start the project, we were quite surprised that the app must be optimized for CLI, which is not a common thing for most of the commercial apps nowadays. 
+Hence, tremendous effort has been put into optimizing the app for the most natural & intuitive User Experience (more details in the Feature Design Consideration). 
+
+To achieve that, the team has had countless debates on how new features should be implemented, and there were even a topic about task's remark that took the team 3 days of debate to settle. 
+
+
+4 days before the deadline, the app didn't have the edit module features, since we knew for sure that an user will almost never need to edit a module's name or code while using the app (since if they entered it wrongly in the first place they could have fixed it right away).
+Yet, because of the strive for perfection, we didn't want to leave a possible situation where "bad" UX happens (Since in the extremely rare case some users will still need to edit that). So we decided to write an additional few hundreds LOC, and did a lot more testing for that new feature (Since EditModule is a very complicated feature to implement)
+
+We don't aim to create an app with a load of features, but aim to create one with just enough features to get all tasks done, and to make our users feel comfortable & happy in the process.
+Our app may look "simple", but **simple is the ultimate sophistication**. 
+
+### Compared to AB3
+Our app can be thought of as a superset of AB3. AB3 was concerned primarily with the management of contacts. We recognised that contacts and friends were a crucial part of any student’s life. Hence, we decided to build upon AB3 by altering the app to be suited for students, and so that the app could help students conveniently manage their student lives.
+
+Hence, we decided to implement module, lesson, and task managers on top of the existing AB3 contacts manager. Now, students will be able to keep track of all their academic and social commitments with a single app, where they previously had to use 4. 
+
+We put in a lot of effort to add these 3 extra features, working late most nights before our v1.2 demo. Furthermore, we decided after that we should integrate all 4 core features seamlessly, to improve upon the existing user experience. As such, for v1.3 users were able to:
+* view all tasks, lessons, and contacts related to a module
+* view all upcoming tasks, and lessons
+* view all future tasks
+Again, this was no easy feat, and our team worked tirelessly to finish our features before v1.3.
```

Comment:
I think the first letter of the bullet points should be capitalised

Reply:
Okay sure